### DISCUSSION TITLE
feature request - suupport multiple BLE interfaces

### DISCUSSION THREAD

**Discussion by lichangche** [hbldh/bleak#1396] [Q&A]
Thanks for the wonderful package! 

Is it possible to support scanning and connecting via multiple BLE interfaces? I.e., hci0, hci1..?  Or has it been supported already? If so, can anyone point  me a link or doc on how to use it? 

Feels like it will fall into a low priority request :) Can we have a discussion on how to implement it so that I might be able to do it myself?

**Comment by dlech** [comment]
I guess it isn't well documented but `BleakScanner` and `BleakClient` take an `adapter` keyword arg. This only works on Linux.

https://bleak.readthedocs.io/en/latest/backends/linux.html?highlight=adapter

This should probably be moved to the `bluez` arg dict since it is platform-specific.

**Reply by lichangche** [reply]
Cool,  thanks. Tried and it worked on raspberry pi.

**Reply by bojanpotocnik** [reply]
Looks like Discussions are not linking when referenced (as Issues and PRs are), so I'll link the comment about using multiple adapters and advertisement callbacks: https://github.com/hbldh/bleak/discussions/867#discussioncomment-8012003
As dlech mentioned, this case is so rare that the workaround probably won't be implemented, so just leaving it here for anyone searching about using multiple adapters [simultaneously].
